fili-core/src/main/java/com/yahoo/bard/webservice/table/PermissiveConcretePhysicalTable.java

```diff
@@ -0,0 +1,40 @@
+// Copyright 2017 Yahoo Inc.
+// Licensed under the terms of the Apache license. Please see LICENSE.md file distributed with this work for terms.
+package com.yahoo.bard.webservice.table;
+
+import com.yahoo.bard.webservice.data.config.names.TableName;
+import com.yahoo.bard.webservice.data.time.ZonedTimeGrain;
+import com.yahoo.bard.webservice.table.availability.PermissiveAvailability;
+
+import java.util.Map;
+import java.util.Set;
+
+import javax.validation.constraints.NotNull;
+
+/**
+ * An implementation of permissive Physical table that is backed by a single fact table.
+ * <p>
+ * This is different from its parent <tt>ConcretePhysicalTable</tt> by using a different
+ * Availability type, which is <tt>PermissiveAvailability</tt>
+ */
+public class PermissiveConcretePhysicalTable extends ConcretePhysicalTable {
+    /**
+     * Create a permissive concrete physical table.
+     *
+     * @param name  Name of the physical table as TableName
+     * @param timeGrain  time grain of the table
+     * @param columns  The columns for this table
+     * @param logicalToPhysicalColumnNames  Mappings from logical to physical names
+     * @param permissiveAvailability the instance of Permissive Availability which makes sure that
+     *                               the table is backed by permissive availability
+     */
+    protected PermissiveConcretePhysicalTable(
```

Comment:
This should be public and does not take a `permissiveAvailability` as its argument, it should construct the availability and pass it to `super`.